### PR TITLE
Backend/create project

### PR DESCRIPTION
**Pull request by oleiade**
This PR adds:
- A Tauri `create_project` command
- A `createProject` helper to call into 👆command from Svelte
- Updates the Sidebar to list projects using the Tauri command
- Adds support for actually creating a project with the "Create a project" button

### CODE REVIEW

File: src-tauri/src/operations.rs [hunk 1]
```diff
@@ -41,12 +42,12 @@ impl LocalProjectManager {
...
 
         // Compute the path of the project in the "projects" directory
-        let project_path = projects_dir.join(name);
+        let project_path = projects_dir.join(&project.name);
 
         // Create the underlying directory for the project
         fs::create_dir(project_path)?;
 
-        Ok(Project::new(name))
+        Ok(project.clone())
```
Comment (Llandy3d) [inline src-tauri/src/operations.rs:50]: I don't think we need the clone here, was there a specific reason ? 🤔

**Review by Llandy3d — APPROVED** [review]

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -57,6 +62,17 @@ async fn list_projects(state: tauri::State<'_, ApplicationState>) -> Result<Vec<
         .map_err(|e| e.to_string())
 }
 
+#[tauri::command]
+async fn create_project(
+    state: tauri::State<'_, ApplicationState>,
+    name: &str,
+    description: Option<&str>,
+) -> Result<models::Project, String> {
```
Comment (going-confetti) [inline src-tauri/src/main.rs:70]: This is related to a comment I just posted in my PR, but I was wondering what would be the best place to store the cloud project id - can it be a part of the `Project` model? If so, it would probably be good to add it as an optional argument here

**Review by going-confetti — APPROVED** [review]
LGTM

File: src/routes/Sidebar.svelte [hunk 3]
```diff
@@ -92,13 +81,18 @@
 								Create a new project
 							</Command.Item>
 						</AlertDialog.Trigger>
```
Comment (allansson) [inline src/routes/Sidebar.svelte:83]: If I click in the project name input, the dialog closes. I think the reason is that focus of the Popover is lost which means it is closed, unmounting the AlertDialog component.

**Review by allansson — APPROVED** [review]
I did find a UI issue. But since it's still possible to create a project and because us frontenders will want to circle back and tighten things up, I'm approving it. 

TLDR; I can create a project. Good job!
